tests/core/test_em_async_js.c

```diff
@@ -1,12 +1,36 @@
 #include <emscripten.h>
 #include <stdio.h>
 
+double result = 0.0;
+
+EMSCRIPTEN_KEEPALIVE
+void set_result() {
+  result += 2.1;
```

Comment:
Seems like you are trying to keep the expected output the same... but you don't need to right?

Maybe just do `+= 1` so the result is more obviously a count of the number of calls?

Also, maybe printf in each call so the sequence is visible in the test output?

Reply:
1 is a integer, though. I think this test intentionally uses doubles which are not ints (to show we return a double properly). I'll add comments (+ printfs as you suggest)

Reply:
I don't see anything in this test to suggest it depends on using doubles.. seems like the test is about the async stuff, not so much the types used.   Maybe I'm missing something?  (If I'm correct then that is another reason not to use doubles.. since it confuses future readers/modifiers as to what the test is actually trying to test).

Reply:
The async code needs to pass through the return type. It has to use a JS double to keep all the valid values possible. That is, if the async code had a `|0` anywhere, or saved the data to HEAP32, it would fail the double part of the test. In the far past I think we had issues there.

Reply:
I don't really get what "fail the double part of the test" means... but it sounds reasonable.   lgtm

Reply:
Failing the double part means that we return an integer. In that case instead of 4.2 we'd print out 4. To pass the test fully we need to handle the result as a double.